rpc_util.go

```diff
@@ -667,14 +668,35 @@ func recv(p *parser, c baseCodec, s *transport.Stream, dc Decompressor, m interf
 	return nil
 }
 
+// Information about RPC
+// Responsible for storing codec, and compressors
+// If stream (s) has  context s.Context which stores rpcInfo that has non nil
+// pointers to codec, and compressors, then we can use preparedMsg for Async message prep
+// and reuse marshalled bytes
 type rpcInfo struct {
 	failfast bool
+	codec    baseCodec
+	cp       Compressor
+	comp     encoding.Compressor
```

Comment:
It seems like these 3 fields are all related to PreparedMsg. Make a new struct to store these and put its pointer into rpcInfo,